run_dir/static/js/projects.js

```diff
@@ -843,21 +843,52 @@ $("#allFields").change(function(e){
   updateTableFields("");
 })
 
+$('#reorganiseReorderbtn').on("click", function() {
+  console.log('hi');
+  $("#tHeaderListul").empty();
+  tHList="";
+  $("#allColFields input[class='filterCheckbox']:checked").each(function(i, elem){
+     tHList+='<li data-name="'+$(this).prop('name')+'">'+$(this).data('displayname')+'</li>';
+   })
+  $('#tHeaderListul').append(tHList);
+});
+
 function updateTableFields(order){
+  var currOrder=new Array();
+  $("#tHeaderListul").find('li').each(function(i,elem){
+    if($(this).data('name'))
+      currOrder.push($(this).data('name'));
+  })
   $("#tHeaderListul").empty();
   tHList="";
   if(order==""){
-    $("#allColFields input[class='filterCheckbox']:checked").each(function(i, elem){
-      tHList+='<li data-name="'+$(this).prop('name')+'">'+$(this).data('displayname')+'</li>';
-    })
+    var checkedItems = $("#allColFields input[class='filterCheckbox']:checked").toArray();
```

Comment:
As we discussed, I think the bulk of this code is not needed. Instead just remove or add the fields corresponding to the boxes that have been unticked/ticked.

Reply:
Changed the logic!